lib/explorer/data_frame.ex

```diff
@@ -1630,11 +1626,27 @@ defmodule Explorer.DataFrame do
         b float [1.0, 4.0, 9.0]
       >
 
+  Different from the macro `mutate/2`, it's possible to "reuse" a variable
+  for different computations:
```

Comment:
I think you can do this as well in queries, it just a bit weird:

```
mutate(df, (
  c = a + b
  [c: c, d: window_sum(c, 2)]
))
```

Reply:
@josevalim I think this is not possible today. It is giving me this error:

```
== Compilation error in file test/explorer/data_frame_test.exs ==
** (CompileError) (for doctest at) lib/explorer/data_frame.ex:1559: cannot invoke remote function Explorer.DataFrame.pull/2 inside a match
    (elixir 1.14.0) src/elixir_fn.erl:17: anonymous fn/4 in :elixir_fn.expand/4
    (stdlib 4.1) lists.erl:1462: :lists.mapfoldl_1/3
    (elixir 1.14.0) src/elixir_fn.erl:23: :elixir_fn.expand/4
    (explorer 0.4.0-dev) expanding macro: Explorer.Query.query/1
    (for doctest at) lib/explorer/data_frame.ex:1558: Explorer.DataFrameTest."doctest Explorer.DataFrame.mutate/2 (135)"/1
    (explorer 0.4.0-dev) expanding macro: Explorer.DataFrame.mutate/2
    (for doctest at) lib/explorer/data_frame.ex:1558: Explorer.DataFrameTest."doctest Explorer.DataFrame.mutate/2 (135)"/1
```

But if I do with `^c`, it works:

```elixir
mutate(df, (
  ^c = a + b
  [c: ^c, d: window_sum(^c, 2)]
))
```

I think the only strange line is the first one. Assign with `^c =` is kind of weird, but the rest makes sense to me.
WDYT?

Reply:
Since it's not a problem with `put/3`, I'm merging this. We can tackle this in another PR.